sdk/common/src/main/java/io/opentelemetry/sdk/common/ScopeConfiguratorBuilder.java

```diff
@@ -0,0 +1,111 @@
+  private final List<Condition<T>> conditions = new ArrayList<>();
+
+  ScopeConfiguratorBuilder(ScopeConfigurator<T> baseScopeConfigurator) {
+    this.baseScopeConfigurator = baseScopeConfigurator;
+  }
+
+  /**
+   * Set the default scope config, which is returned by {@link ScopeConfigurator#apply(Object)} if a
+   * {@link InstrumentationScopeInfo} does not match any {@link #addCondition(Predicate, Object)
+   * conditions}. If a default is not set, an SDK defined default is used.
+   */
+  public ScopeConfiguratorBuilder<T> setDefault(T defaultScopeConfig) {
+    this.defaultScopeConfig = defaultScopeConfig;
+    return this;
+  }
+
+  /**
+   * Add a condition. Conditions are evaluated in order. The {@code scopeConfig} for the first match
+   * is returned by {@link ScopeConfigurator#apply(Object)}.
+   *
+   * @param scopePredicate predicate that {@link InstrumentationScopeInfo}s are evaluated against
+   * @param scopeConfig the scope config to use when this condition is the first matching {@code
+   *     scopePredicate}
+   * @see #nameMatchesGlob(String)
+   * @see #nameEquals(String)
+   */
+  public ScopeConfiguratorBuilder<T> addCondition(
+      Predicate<InstrumentationScopeInfo> scopePredicate, T scopeConfig) {
+    conditions.add(new Condition<>(scopePredicate, scopeConfig));
+    return this;
+  }
+
+  /**
+   * Helper function for pattern matching {@link InstrumentationScopeInfo#getName()} against the
+   * {@code globPattern}.
+   *
+   * <p>{@code globPattern} may contain the wildcard characters {@code *} and {@code ?} with the
+   * following matching criteria:
+   *
+   * <ul>
+   *   <li>{@code *} matches 0 or more instances of any character
+   *   <li>{@code ?} matches exactly one instance of any character
+   * </ul>
+   *
+   * @see #addCondition(Predicate, Object)
+   */
+  public static Predicate<InstrumentationScopeInfo> nameMatchesGlob(String globPattern) {
+    Predicate<String> globPredicate = GlobUtil.toGlobPatternPredicate(globPattern);
+    return scopeInfo -> globPredicate.test(scopeInfo.getName());
+  }
+
+  /**
+   * Helper function for exact matching {@link InstrumentationScopeInfo#getName()} against the
+   * {@code scopeName}.
+   *
+   * @see #addCondition(Predicate, Object)
+   */
+  public static Predicate<InstrumentationScopeInfo> nameEquals(String scopeName) {
+    return scopeInfo -> scopeInfo.getName().equals(scopeName);
+  }
+
+  /** Build a {@link ScopeConfigurator} with the configuration of this builder. */
+  public ScopeConfigurator<T> build() {
+    // TODO: return an instance with toString implementation which self describes rules
+    return scopeInfo -> {
+      T scopeConfig = baseScopeConfigurator.apply(scopeInfo);
+      if (scopeConfig != null) {
+        return scopeConfig;
+      }
+      for (Condition<T> condition : conditions) {
+        if (condition.scopeMatcher.test(scopeInfo)) {
+          return condition.scopeConfig;
+        }
+      }
+      return defaultScopeConfig;
```

Comment:
Oh, I think I've answered my own question -- it's defaulted per signal I think.

Reply:
Yeah the default is per signal. 

And all the signals will initially be aligned, since initially all consist of only the enabled / disabled property which defaults to true. But as signal-specific scope config options are introduced, it will be important for each signal to be able to individually define their default.